### DISCUSSION TITLE
change glsl shader position/order

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14203] [Q&A] [answered]
Hi, I would like to apply my Vibrance shader at the last possible position/order in the glsl shader pipeline. Is this possible?
Here it seems to be at the beginning, I would like to put it after "transfer function (srgb)"

I activate/toggle this shader with a keybind in the input.conf file `e change-list glsl-shaders toggle "~~/shaders/vibrance.glsl"`
![glsl order dd](https://github.com/mpv-player/mpv/assets/139760266/8f9100eb-e4a0-4674-b3dc-df2bd8750203)

Thank you :)

**Comment by kasper93** [comment]
Take a look at this. https://github.com/mpv-player/mpv/wiki/Video-output---shader-stage-diagram It may not be fully up to date with gpu-next, but will give you idea what are the stages and how HOOK will position your shader. Take a look at HOOK documentation too https://libplacebo.org/custom-shaders/

**Reply by geextahslex** [reply]
Thanks for your reply but this is a bit beyond my comprehension ^^

**Reply by kasper93** [reply, marked as answer]
You probably have `HOOK LUMA` currently, if you want to move it before dithering, `HOOK OUTPUT` is the one you should use. Note the input/output of the shader is different in this case, OUTPUT is RGB, while luma is only Y.

**Reply by geextahslex** [reply]
Okay it was `HOOK MAIN`, now it is in the right place, but I think it behaves different from what I hoped ^^'
Thank you anyway :)

also interesting that the usage of the shader dropped from 14% to 4% on HDR. On SDR it is still 14%
